panther_lights/src/controller_node.py

```diff
@@ -21,10 +20,89 @@
 
 
 class PantherAnimation:
+    ANIMATION_DEFAULT_PRIORITY = 3
+    ANIMATION_DEFAULT_TIMEOUT = 120
+
     front: Animation
     rear: Animation
-    interrupting = False
-    repeating = False
+    name: str = 'NAME_NOT_DEFINED'
+    priority: int = ANIMATION_DEFAULT_PRIORITY
+    timeout: float = ANIMATION_DEFAULT_TIMEOUT
+    repeating: bool = False
+
+    def __init__(self) -> None:
+        self._init_time = rospy.get_time()
+
+    @property
+    def init_time(self) -> float:
+        return self._init_time
+
+
+class AnimationsQueue:
+    def __init__(self, max_queue_size) -> None:
+        self._queue = []
+        self._max_queue_size = max_queue_size
+
+    def put(self, animation: PantherAnimation, put_front: int = False) -> None:
+        self.validate_queue()
+
+        if animation.priority == 1:
+            self.clear()
+
+        if len(self._queue) == self._max_queue_size:
+            rospy.logwarn(f'{rospy.get_name()} Animation queue overloaded')
+            return
+
+        if put_front:
+            self._queue.insert(0, animation)
+        else:
+            self._queue.append(animation)
+        self._queue.sort(key=self._get_priority)
```

Comment:
```suggestion
        self._queue.sort(key=lambda x.priority)
```
I think this would be a better way. Also aren't we planning to implement other option to for example sort by time left for animation as well?